### ISSUE TITLE
CuDNNLSTM and LSTM give different results

### ISSUE THREAD

**Issue by GarrisonD** [keras-team/keras#13399]
**System information**  
- Have I written custom code (as opposed to using example directory):  no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Google Colab
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.14.0
- Keras version:  2.2.5
- Python version:  3
- CUDA/cuDNN version: provided by Google Colab
- GPU model and memory: provided by Google Colab

**Describe the current behavior**

`CuDNNLSTM` doesn't convergence as good as `LSTM`.

**Describe the expected behavior**

`CuDNNLSTM` convergences as good as `LSTM`.

**Code to reproduce the issue**

```python3
import numpy as np
from keras.models import Sequential
from keras.layers import LSTM, Dense

X = np.array([[10, 20, 30],
              [20, 30, 40],
              [30, 40, 50],
              [40, 50, 60]]).reshape(-1, 3, 1)

y = np.array([40,
              50,
              60,
              70])

model = Sequential()
model.add(LSTM(50, activation='relu', input_shape=(3, 1)))
model.add(Dense(1))

model.compile(optimizer='adam', loss='mse')
model.fit(X, y, epochs=100, verbose=2)

model.predict(np.array([50, 60, 70]).reshape(-1, 3, 1), verbose=0)
```

When `LSTM` is used then the training loss is decreasing from 3K+ to 2.0 and the model predicts values close to expected. The expected value is 80. See output below:

```
Epoch 1/100
 - 1s - loss: 3491.9238
Epoch 2/100
 - 0s - loss: 3438.5774
Epoch 3/100
 - 0s - loss: 3384.2285
Epoch 4/100
 - 0s - loss: 3329.1284
Epoch 5/100
 - 0s - loss: 3273.1770
Epoch 6/100
 - 0s - loss: 3216.3093
Epoch 7/100
 - 0s - loss: 3162.0608
Epoch 8/100
 - 0s - loss: 3107.3906
Epoch 9/100
 - 0s - loss: 3054.1001
Epoch 10/100
 - 0s - loss: 3000.5264
Epoch 11/100
 - 0s - loss: 2946.1025
Epoch 12/100
 - 0s - loss: 2891.5835
Epoch 13/100
 - 0s - loss: 2836.6155
Epoch 14/100
 - 0s - loss: 2781.3086
Epoch 15/100
 - 0s - loss: 2725.3088
Epoch 16/100
 - 0s - loss: 2668.4060
Epoch 17/100
 - 0s - loss: 2610.8018
Epoch 18/100
 - 0s - loss: 2552.2517
Epoch 19/100
 - 0s - loss: 2490.9800
Epoch 20/100
 - 0s - loss: 2426.2964
Epoch 21/100
 - 0s - loss: 2359.6975
Epoch 22/100
 - 0s - loss: 2290.8906
Epoch 23/100
 - 0s - loss: 2220.0054
Epoch 24/100
 - 0s - loss: 2146.4326
Epoch 25/100
 - 0s - loss: 2069.2437
Epoch 26/100
 - 0s - loss: 1988.9077
Epoch 27/100
 - 0s - loss: 1904.7368
Epoch 28/100
 - 0s - loss: 1816.5853
Epoch 29/100
 - 0s - loss: 1726.2642
Epoch 30/100
 - 0s - loss: 1635.0527
Epoch 31/100
 - 0s - loss: 1540.0222
Epoch 32/100
 - 0s - loss: 1443.5552
Epoch 33/100
 - 0s - loss: 1347.0227
Epoch 34/100
 - 0s - loss: 1249.4344
Epoch 35/100
 - 0s - loss: 1149.9791
Epoch 36/100
 - 0s - loss: 1050.3970
Epoch 37/100
 - 0s - loss: 951.8585
Epoch 38/100
 - 0s - loss: 856.0686
Epoch 39/100
 - 0s - loss: 765.4115
Epoch 40/100
 - 0s - loss: 672.0874
Epoch 41/100
 - 0s - loss: 584.8623
Epoch 42/100
 - 0s - loss: 499.0566
Epoch 43/100
 - 0s - loss: 418.8430
Epoch 44/100
 - 0s - loss: 344.9994
Epoch 45/100
 - 0s - loss: 279.1168
Epoch 46/100
 - 0s - loss: 220.1147
Epoch 47/100
 - 0s - loss: 169.5065
Epoch 48/100
 - 0s - loss: 129.6169
Epoch 49/100
 - 0s - loss: 101.8518
Epoch 50/100
 - 0s - loss: 85.8431
Epoch 51/100
 - 0s - loss: 81.4507
Epoch 52/100
 - 0s - loss: 84.7548
Epoch 53/100
 - 0s - loss: 93.1215
Epoch 54/100
 - 0s - loss: 102.8541
Epoch 55/100
 - 0s - loss: 112.9857
Epoch 56/100
 - 0s - loss: 118.0103
Epoch 57/100
 - 0s - loss: 116.8881
Epoch 58/100
 - 0s - loss: 112.3035
Epoch 59/100
 - 0s - loss: 104.7622
Epoch 60/100
 - 0s - loss: 95.4420
Epoch 61/100
 - 0s - loss: 85.1529
Epoch 62/100
 - 0s - loss: 74.6678
Epoch 63/100
 - 0s - loss: 64.9176
Epoch 64/100
 - 0s - loss: 56.4074
Epoch 65/100
 - 0s - loss: 48.9455
Epoch 66/100
 - 0s - loss: 42.7875
Epoch 67/100
 - 0s - loss: 38.0182
Epoch 68/100
 - 0s - loss: 34.0546
Epoch 69/100
 - 0s - loss: 30.7339
Epoch 70/100
 - 0s - loss: 27.8845
Epoch 71/100
 - 0s - loss: 25.2191
Epoch 72/100
 - 0s - loss: 22.8869
Epoch 73/100
 - 0s - loss: 20.8358
Epoch 74/100
 - 0s - loss: 18.7593
Epoch 75/100
 - 0s - loss: 16.7336
Epoch 76/100
 - 0s - loss: 14.8623
Epoch 77/100
 - 0s - loss: 13.1597
Epoch 78/100
 - 0s - loss: 11.6585
Epoch 79/100
 - 0s - loss: 10.4835
Epoch 80/100
 - 0s - loss: 9.6534
Epoch 81/100
 - 0s - loss: 9.0795
Epoch 82/100
 - 0s - loss: 8.5818
Epoch 83/100
 - 0s - loss: 7.9778
Epoch 84/100
 - 0s - loss: 7.3678
Epoch 85/100
 - 0s - loss: 6.9376
Epoch 86/100
 - 0s - loss: 6.6947
Epoch 87/100
 - 0s - loss: 6.2735
Epoch 88/100
 - 0s - loss: 5.8172
Epoch 89/100
 - 0s - loss: 5.3292
Epoch 90/100
 - 0s - loss: 4.8252
Epoch 91/100
 - 0s - loss: 4.2926
Epoch 92/100
 - 0s - loss: 3.7602
Epoch 93/100
 - 0s - loss: 3.2560
Epoch 94/100
 - 0s - loss: 2.8064
Epoch 95/100
 - 0s - loss: 2.4363
Epoch 96/100
 - 0s - loss: 2.2590
Epoch 97/100
 - 0s - loss: 2.1824
Epoch 98/100
 - 0s - loss: 2.0952
Epoch 99/100
 - 0s - loss: 1.9794
Epoch 100/100
 - 0s - loss: 1.8399

array([[84.035324]], dtype=float32)
```

***

```python3
import numpy as np
from keras.models import Sequential
from keras.layers import Activation, CuDNNLSTM, Dense

X = np.array([[10, 20, 30],
              [20, 30, 40],
              [30, 40, 50],
              [40, 50, 60]]).reshape(-1, 3, 1)

y = np.array([40,
              50,
              60,
              70])

model = Sequential()
model.add(CuDNNLSTM(50, input_shape=(3, 1)))
model.add(Activation('relu'))
model.add(Dense(1))

model.compile(optimizer='adam', loss='mse')
model.fit(X, y, epochs=100, verbose=2)

model.predict(np.array([50, 60, 70]).reshape(-1, 3, 1), verbose=0)
```

When `LSTM` is replaced by `CuDNNLSTM` then the training loss is decreasing from 3K+ to only 2.5K and the model predicts values not close to expected at all - from 1 till 6 or so (instead of 80):

```
Epoch 1/100
 - 1s - loss: 3145.2993
Epoch 2/100
 - 0s - loss: 3141.6865
Epoch 3/100
 - 0s - loss: 3138.1990
Epoch 4/100
 - 0s - loss: 3134.8298
Epoch 5/100
 - 0s - loss: 3131.5698
Epoch 6/100
 - 0s - loss: 3128.4106
Epoch 7/100
 - 0s - loss: 3125.3457
Epoch 8/100
 - 0s - loss: 3122.3694
Epoch 9/100
 - 0s - loss: 3119.4763
Epoch 10/100
 - 0s - loss: 3116.6616
Epoch 11/100
 - 0s - loss: 3113.9209
Epoch 12/100
 - 0s - loss: 3111.2515
Epoch 13/100
 - 0s - loss: 3108.6619
Epoch 14/100
 - 0s - loss: 3106.1387
Epoch 15/100
 - 0s - loss: 3103.6606
Epoch 16/100
 - 0s - loss: 3101.2271
Epoch 17/100
 - 0s - loss: 3098.8389
Epoch 18/100
 - 0s - loss: 3096.4985
Epoch 19/100
 - 0s - loss: 3094.2026
Epoch 20/100
 - 0s - loss: 3091.9460
Epoch 21/100
 - 0s - loss: 3089.7190
Epoch 22/100
 - 0s - loss: 3087.5164
Epoch 23/100
 - 0s - loss: 3085.3325
Epoch 24/100
 - 0s - loss: 3083.1611
Epoch 25/100
 - 0s - loss: 3080.9958
Epoch 26/100
 - 0s - loss: 3078.8315
Epoch 27/100
 - 0s - loss: 3076.6611
Epoch 28/100
 - 0s - loss: 3074.4736
Epoch 29/100
 - 0s - loss: 3072.2612
Epoch 30/100
 - 0s - loss: 3070.0164
Epoch 31/100
 - 0s - loss: 3067.7285
Epoch 32/100
 - 0s - loss: 3065.3870
Epoch 33/100
 - 0s - loss: 3062.9819
Epoch 34/100
 - 0s - loss: 3060.5022
Epoch 35/100
 - 0s - loss: 3057.9590
Epoch 36/100
 - 0s - loss: 3055.3438
Epoch 37/100
 - 0s - loss: 3052.6143
Epoch 38/100
 - 0s - loss: 3049.7588
Epoch 39/100
 - 0s - loss: 3046.8049
Epoch 40/100
 - 0s - loss: 3043.7068
Epoch 41/100
 - 0s - loss: 3040.4429
Epoch 42/100
 - 0s - loss: 3037.0476
Epoch 43/100
 - 0s - loss: 3033.4646
Epoch 44/100
 - 0s - loss: 3029.7166
Epoch 45/100
 - 0s - loss: 3025.7803
Epoch 46/100
 - 0s - loss: 3021.6904
Epoch 47/100
 - 0s - loss: 3017.3853
Epoch 48/100
 - 0s - loss: 3012.8945
Epoch 49/100
 - 0s - loss: 3008.2705
Epoch 50/100
 - 0s - loss: 3003.4917
Epoch 51/100
 - 0s - loss: 2998.6362
Epoch 52/100
 - 0s - loss: 2993.7434
Epoch 53/100
 - 0s - loss: 2988.7468
Epoch 54/100
 - 0s - loss: 2983.6516
Epoch 55/100
 - 0s - loss: 2978.4775
Epoch 56/100
 - 0s - loss: 2973.2451
Epoch 57/100
 - 0s - loss: 2967.9773
Epoch 58/100
 - 0s - loss: 2962.6978
Epoch 59/100
 - 0s - loss: 2957.4329
Epoch 60/100
 - 0s - loss: 2952.2104
Epoch 61/100
 - 0s - loss: 2947.0593
Epoch 62/100
 - 0s - loss: 2942.0093
Epoch 63/100
 - 0s - loss: 2937.0874
Epoch 64/100
 - 0s - loss: 2932.3191
Epoch 65/100
 - 0s - loss: 2927.7231
Epoch 66/100
 - 0s - loss: 2923.3101
Epoch 67/100
 - 0s - loss: 2919.0811
Epoch 68/100
 - 0s - loss: 2915.0286
Epoch 69/100
 - 0s - loss: 2911.1357
Epoch 70/100
 - 0s - loss: 2907.3804
Epoch 71/100
 - 0s - loss: 2903.7349
Epoch 72/100
 - 0s - loss: 2900.1709
Epoch 73/100
 - 0s - loss: 2896.6619
Epoch 74/100
 - 0s - loss: 2893.1836
Epoch 75/100
 - 0s - loss: 2889.7104
Epoch 76/100
 - 0s - loss: 2886.2241
Epoch 77/100
 - 0s - loss: 2882.7104
Epoch 78/100
 - 0s - loss: 2879.1570
Epoch 79/100
 - 0s - loss: 2875.5579
Epoch 80/100
 - 0s - loss: 2871.9116
Epoch 81/100
 - 0s - loss: 2868.2212
Epoch 82/100
 - 0s - loss: 2864.4956
Epoch 83/100
 - 0s - loss: 2860.7515
Epoch 84/100
 - 0s - loss: 2857.0112
Epoch 85/100
 - 0s - loss: 2853.3037
Epoch 86/100
 - 0s - loss: 2849.6602
Epoch 87/100
 - 0s - loss: 2846.1116
Epoch 88/100
 - 0s - loss: 2842.6846
Epoch 89/100
 - 0s - loss: 2839.3984
Epoch 90/100
 - 0s - loss: 2836.2637
Epoch 91/100
 - 0s - loss: 2833.2822
Epoch 92/100
 - 0s - loss: 2830.4485
Epoch 93/100
 - 0s - loss: 2827.7524
Epoch 94/100
 - 0s - loss: 2825.1802
Epoch 95/100
 - 0s - loss: 2822.7170
Epoch 96/100
 - 0s - loss: 2820.3481
Epoch 97/100
 - 0s - loss: 2818.0605
Epoch 98/100
 - 0s - loss: 2815.8418
Epoch 99/100
 - 0s - loss: 2813.6821
Epoch 100/100
 - 0s - loss: 2811.5723

array([[3.1825027]], dtype=float32)
```

**Comment by baggyg** [comment]
You could see how CuDNNLSTM does if you let it learn longer (I.e. more epochs). It doesn't look like it has reached the bottom of the gradient descent. However I've seen similar results to these. On a simple 1 layer network CuDNNLSTM is less accurate regardless of the epochs / learning rate etc. 

It would be nice to know why the two differ in terms of result. Is something being sacrificed for speed?

**Comment by georgesterpu** [comment]
Are you sure that `LSTM(activation='relu')` is the same thing as `LSTM(); Activation('relu')` ?
